jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/AdministrationControllerTest.groovy

```diff
@@ -109,12 +107,117 @@ class AdministrationControllerTest extends Specification {
             groupName           | groupDescription    |errorMessage                                                    | caseName
             ''                  | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name is empty'
             '     '             | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name contains only spaces'
             alphanumeric(101)   | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name too long'
             alphanumeric(1,100) | alphanumeric(256)   |"Group description length must be between 0 and 255 characters" | 'Group description too long'
     }
+
+    def 'must not be able to edit group without admin permission'() {
+        given: 'User signed in and forum has some existing group'
+            def session = users.signInAsRegisteredUser(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+        when: 'User attempts to edit an existing group'
+            def dto = new GroupAdministrationDto()
+            dto.setId(savedGroupId)
+            dto.setName(alphanumeric(GROUP_NAME_MAX_LENGTH))
+            dto.setDescription(alphanumeric(GROUP_DESCRIPTION_MAX_LENGTH))
+            groups.edit(dto, session)
+        then:
+            thrown(AccessDeniedException)
+    }
+
+    def 'test edit group success'() {
+        given: 'Admin signed in and forum has some existing group'
+            def session = users.signInAsAdmin(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+        when: 'Admin attempts to edit an existing group'
+            def dto = new GroupAdministrationDto()
+            dto.setId(savedGroupId)
+            dto.setName(alphanumeric(GROUP_NAME_MAX_LENGTH))
+            dto.setDescription(alphanumeric(GROUP_DESCRIPTION_MAX_LENGTH))
+            groups.edit(dto, session)
+        then:
+            groups.isExist(dto.name)
+        and: 'All fields of edited group (except ID) has been updated'
+            def edited = groupsService.getById(savedGroupId)
+            savedGroupId == edited.id
+            dto.name == edited.name
+            dto.description == edited.description
+    }
+
+    def 'must not be able to edit not existing group'() {
+        given: 'Admin signed in and forum has some existing group'
+            def session = users.signInAsAdmin(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+        when: 'Admin attempts to edit not existing group'
+            def dto = new GroupAdministrationDto()
+            dto.setId(savedGroupId + 1)
+            dto.setName(alphanumeric(GROUP_NAME_MAX_LENGTH))
+            dto.setDescription(alphanumeric(GROUP_DESCRIPTION_MAX_LENGTH))
+            groups.edit(dto, session)
+        then:
+            thrown(WrongResponseException)
+    }
+
+    def 'after changing group name must not be able to find group by old name'() {
```

Comment:
Do you think this test is really needed? It's hard to imagine a code that would pass the previous tests while failing this one..